Assets/Scripts/Utils/ObjectPool.cs

```diff
@@ -0,0 +1,38 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ObjectPool<T> where T : MonoBehaviour
+{
+    private T template;
+    private List<T> pool;
+    private int nextIndex = 0;
+    private int capacity;
+
+    public ObjectPool(T template, int capacity = 70)
+    {
+        this.template = template;
+        this.capacity = capacity;
+        pool = new List<T>(capacity);
+    }
+
+    public T Get()
```

Comment:
It would be really funny to name this method `Surface` to go with the pooling name, but I Understand why we would opt for a name that easier to parse😅

Reply:
Thought of `Pull` myself, but I felt we should go for something braindead